gapic/schema/wrappers.py

```diff
@@ -767,6 +767,29 @@ def http_opt(self) -> Optional[Dict[str, str]]:
         # TODO(yon-mg): enums for http verbs?
         return answer
 
+    @property
+    def path_params(self) -> Sequence[str]:
+        """Return the path parameters found in the http annotation url"""
+        # TODO(yon-mg): fully implement grpc transcoding (currently only handles basic case)
+        if self.http_opt is None:
+            return []
+        pattern = r'\{\w+\}'
+
+        return [x[1:-1] for x in re.findall(pattern, self.http_opt['url'])]
+
+    @property
+    def query_params(self) -> Set[str]:
+        """Return query parameters for API call as determined by http annotation and grpc transcoding"""
```

Comment:
!! Q: can you know all the query parameters ahead of time? If a user sets a field nested inside a message that is not listed in the path template nor in the body part of the annotation, it should be reported as a query param, but presumably in dotted notation (see my other comment about dotted notation being used in practice for the body even though disallowed in theory).

Reply:
This is an oversight yes. Due to these nested fields I don't think it makes sense to identify all query params ahead of time, though it is possible! I am moving this logic out of the generator entirely and have left appropriate todos.

Reply:
OK. I see you added TODOs to come back to this later. sgtm.